superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FilterTabTitle.tsx

```diff
@@ -0,0 +1,196 @@
+  isRemoved: boolean;
+  index: number;
+  getFilterTitle: (id: string) => string;
+  restoreFilter: (id: string) => void;
+  onRearrage: (itemId: string, targetIndex: number) => void;
+  onRemove: (id: string) => void;
+}
+
+interface DragItem {
+  index: number;
+  id: string;
+  type: string;
+}
+
+export const FilterTabTitle: React.FC<FilterTabTitleProps> = ({
+  id,
+  index,
+  isRemoved,
+  getFilterTitle,
+  restoreFilter,
+  onRearrage,
+  onRemove,
+}) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const [{ isDragging }, drag] = useDrag({
+    item: { id, type: FILTER_TYPE, index },
+    collect: (monitor: any) => ({
+      isDragging: monitor.isDragging(),
+    }),
+  });
+  const [, drop] = useDrop({
+    accept: FILTER_TYPE,
+    hover: (item: DragItem, monitor: DropTargetMonitor) => {
+      if (!ref.current) {
+        return;
+      }
+
+      const dragIndex = item.index;
+      const hoverIndex = index;
+
+      // Don't replace items with themselves
+      if (dragIndex === hoverIndex) {
+        return;
+      }
+      // Determine rectangle on screen
+      const hoverBoundingRect = ref.current?.getBoundingClientRect();
+
+      // Get vertical middle
+      const hoverMiddleY =
+        (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+      // Determine mouse position
+      const clientOffset = monitor.getClientOffset();
+
+      // Get pixels to the top
+      const hoverClientY = (clientOffset as XYCoord).y - hoverBoundingRect.top;
+
+      // Only perform the move when the mouse has crossed half of the items height
+      // When dragging downwards, only move when the cursor is below 50%
+      // When dragging upwards, only move when the cursor is above 50%
+
+      // Dragging downwards
+      if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+        return;
+      }
+
+      // Dragging upwards
+      if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+        return;
+      }
+
+      onRearrage(item.id, index);
+      // Note: we're mutating the monitor item here.
+      // Generally it's better to avoid mutations,
+      // but it's good here for the sake of performance
+      // to avoid expensive index searches.
+      // eslint-disable-next-line no-param-reassign
+      item.index = hoverIndex;
+    },
+  });
+  drag(drop(ref));
+  return (
+    <TabTitleContainer
+      ref={ref}
+      style={{
+        opacity: isDragging ? 0.5 : 1,
+      }}
+    >
+      <StyledFilterTabTitle className={isRemoved ? 'removed' : ''}>
+        <StyledFilterTitle>
+          {isRemoved ? t('(Removed)') : getFilterTitle(id)}
+        </StyledFilterTitle>
+        {isRemoved && (
+          <StyledSpan
+            role="button"
+            data-test="undo-button"
+            tabIndex={0}
+            onClick={() => restoreFilter(id)}
+          >
+            {t('Undo?')}
+          </StyledSpan>
+        )}
+      </StyledFilterTabTitle>
+      <div style={{ alignSelf: 'flex-end' }}>
```

Comment:
Avoid using `style`. Currently, we are using Emotion `styled` or `css` depending on the situation. You can check this [guide](https://github.com/apache/superset/issues/15264) for more information.